marlowe/test/Spec/Marlowe/Semantics/Arbitrary.hs

```diff
@@ -419,6 +420,35 @@ choiceNotInBounds bounds =
     suchThat arbitrary $ \chosenNum -> not $ any (inBound chosenNum) bounds
 
 
+-- | Generate relevant Input content for a given input action
+-- interestingInput :: Bool -> Action -> [InputContent]
+-- interestingInput validity (Choice choiceId bounds) =
+--   map (IChoice choiceId) (interestingChoiceNum bounds)
+--   ++ map (\choiceId -> IChoice choiceId bounds) (interestingChoiceId choiceId)
+-- interestingInput validity (Choice choiceId bounds) = IChoice choiceId <$> interestingChoiceNum' validity bounds
+
+-- interestingChoiceId :: ChoiceId -> [ChoiceId]
+-- interestingChoiceId _ = error "not implemented yet"
+
+-- interestingParties :: Party -> [Party]
+-- interestingParties (Role "Alice") = [Role "Alice", Role "Alic", Role "Alices"]
+
+-- interestingChoiceNum :: [Bound] -> [ChosenNum]
+-- interestingChoiceNum bounds = concatMap interestingChoiceNum' True bounds
+
+interestingChoiceNum' :: Bool -> Bound -> [ChosenNum]
+interestingChoiceNum' True (Bound lower upper)  = validValues lower upper
+interestingChoiceNum' False (Bound lower upper) = invalidValues lower upper
+
+validValues :: Integer -> Integer -> [Integer]
+validValues lower upper = [x | x <- testValueRange, x >= lower  && x <= upper]
+
+invalidValues :: Integer -> Integer -> [Integer]
+invalidValues lower upper = [x | x <- testValueRange, x < lower  || x > upper]
+
+testValueRange :: [Integer]
+testValueRange = [(-10000)..10000]
```

Comment:
hey @bwbush I think I found a cleaner solution to this problem, thanks to your suggestion of thinking about this problem is a number line. However, this code is still dependent on a hard-coded number range. Is there a better way to solve this problem? Maybe we can circle back to this when it starts failing?